core/src/main/java/org/neo4j/graphalgo/core/utils/paged/HugeIntArray.java

```diff
@@ -0,0 +1,476 @@
+            int[] page = new int[intSize];
+
+            tracker.add(shallowSizeOfInstance(SingleHugeIntArray.class));
+            tracker.add(sizeOfIntArray(intSize));
+
+            return new SingleHugeIntArray(intSize, page);
+        }
+
+        private final int size;
+        private int[] page;
+
+        private SingleHugeIntArray(int size, int[] page) {
+            this.size = size;
+            this.page = page;
+        }
+
+        @Override
+        public int get(long index) {
+            assert index < size;
+            return page[(int) index];
+        }
+
+        @Override
+        public void set(long index, int value) {
+            assert index < size;
+            page[(int) index] = value;
+        }
+
+        @Override
+        public void or(long index, int value) {
+            assert index < size;
+            page[(int) index] |= value;
+        }
+
+        @Override
+        public int and(long index, int value) {
+            assert index < size;
+            return page[(int) index] &= value;
+        }
+
+        @Override
+        public void addTo(long index, int value) {
+            assert index < size;
+            page[(int) index] += value;
+        }
+
+        @Override
+        public void setAll(LongToIntFunction gen) {
+            Arrays.setAll(page, gen::applyAsInt);
+        }
+
+        @Override
+        public void fill(int value) {
+            Arrays.fill(page, value);
+        }
+
+        @Override
+        public void copyTo(HugeIntArray dest, long length) {
+            if (length > size) {
+                length = size;
+            }
+            if (length > dest.size()) {
+                length = dest.size();
+            }
+            if (dest instanceof SingleHugeIntArray) {
+                SingleHugeIntArray dst = (SingleHugeIntArray) dest;
+                System.arraycopy(page, 0, dst.page, 0, (int) length);
+                Arrays.fill(dst.page, (int) length, dst.size, 0);
+            } else if (dest instanceof PagedHugeIntArray) {
+                PagedHugeIntArray dst = (PagedHugeIntArray) dest;
+                int start = 0;
+                int remaining = (int) length;
+                for (int[] dstPage : dst.pages) {
+                    int toCopy = Math.min(remaining, dstPage.length);
+                    if (toCopy == 0) {
+                        Arrays.fill(page, 0);
+                    } else {
+                        System.arraycopy(page, start, dstPage, 0, toCopy);
+                        if (toCopy < dstPage.length) {
+                            Arrays.fill(dstPage, toCopy, dstPage.length, 0);
+                        }
+                        start += toCopy;
+                        remaining -= toCopy;
+                    }
+                }
+            }
+        }
+
+        @Override
+        public long size() {
+            return size;
+        }
+
+        @Override
+        public long release() {
+            if (page != null) {
```

Comment:
I think I wanted to not drag the Tracker around in the arrays, but I agree that it is inconsistent with what other release methods do. I'll change it so that the tracker is a parameter of the release method